conformance/lang/expressions/is-expr/is_expr.balt

```diff
@@ -0,0 +1,1767 @@
+    io:println(g["d"] is map<anydata> & readonly); // @output true
+}
+
+Test-Case: output
+Description: Test json type as type descriptor in type test expression.
+Labels: is-expr, module-type-defn, record-type, mapping-constructor-expr, json, string, int, decimal, float,
+        union-type, map-type, member-access-expr
+
+type Record1 record {
+    int i;
+};
+
+function init() {
+    Record1 a = {i: 100};
+    io:println(a is json); // @output false
+    io:println(a.i is json); // @output true
+
+    string? b = "ABC";
+    io:println(b is json); // @output true
+
+    int|string|decimal|float c = "ABC";
+    io:println(c is json); // @output true
+
+    map<json> d = {
+        a: 1,
+        b: "foo",
+        c: null,
+        d: {
+            a1: null,
+            b1: true
+        }
+    };
+    io:println(d is json); // @output true
+    io:println(d["a"] is json); // @output true
+    io:println(d["b"] is json); // @output true
+    io:println(d["c"] is json); // @output true
+    io:println(d["d"] is json); // @output true
+}
+
+Test-Case: output
+Description: Test byte type as type descriptor in type test expression.
+Labels: is-expr, byte, int, array-type, list-constructor-expr, int:Signed32, int:Signed16, int:Unsigned32,
+        int:Unsigned16, int:Unsigned8, union-type
+
+function init() {
+    int a = 10;
+    io:println(a is byte); // @output true
+
+    int[] b = [1, 2, 3];
+    io:println(b[2] is byte); // @output true
+
+    int:Signed32 c = 10;
+    io:println(c is byte); // @output true
+
+    int:Signed16 d = 10;
+    io:println(d is byte); // @output true
+
+    int:Unsigned32 e = 10;
+    io:println(e is byte); // @output true
+
+    int:Unsigned16 f = 10;
+    io:println(f is byte); // @output true
+
+    int:Unsigned16 g = 10;
+    io:println(g is byte); // @output true
+
+    int:Unsigned8 h = 10;
+    io:println(h is byte); // @output true
+
+    int:Unsigned16|int:Signed8 i = 10;
+    io:println(i is byte); // @output true
+}
+
+Test-Case: output
+Description: Test type reference as type descriptor in type test expression.
+Labels: is-expr, module-const-decl, int, string, union-type, record-type, table-type, mapping-constructor-expr,
+
+const ONE = 1;
+const JOHN = "john";
+
+type A int|string;
+
+type B ONE|JOHN;
+
+type C record {|
+    string...;
+|};
+
+type D record {
+    readonly string a;
+    string b;
+    string c?;
+};
+
+type E table<D>;
+
+type F table<D> key(a);
+
+type G table<D> key<string>;
+
+function init() {
+    io:println(1 is A); // @output true
+    io:println("Mike" is B); // @output false
+
+    record {|string a; string b;|} a = {a: "A", b: "B"};
+    io:println(a is C); // @output true
+    io:println(a is D); // @output false
+
+    io:println(table key(a) [{a: "A", b: "B"}] is E); // @output true
+
+    F b = table [{a: "A", b: "B"}];
+    io:println(b is E); // @output true
+    io:println(b is G); // @output true
+}
+
+Test-Case: output
```

Comment:
Shall we add few other scenarios type test is being used.
eg: queries, elvis...